fix: provide iife build to prevent conflicts with AMD (#274)
algolia/search-insights.js: README.md

| `cookieDuration`  | `number`       | `15552000000` (6 months) | The cookie duration in milliseconds            |
| `userToken`       | `string`       | `undefined` (optional)   | Initial userToken. When given, anonymous userToken will not be set. |

**Note for Require.js users**:

> If using [Require.js](https://requirejs.org/), the default build of the library being a UMD build, can conflict. You may experience this issue known as [Mismatched anonymous define() modules](https://requirejs.org/docs/errors.html#mismatch).

Reviewer: ```suggestion
When using [Require.js](https://requirejs.org/), the default UMD build might conflict and throw with a "Mismatched anonymous define() modules" message. This is a [known Require.js issue](https://requirejs.org/docs/errors.html#mismatch).
```

> To work around this problem, avoid loading the library via a `define()` call. Instead, change `ALGOLIA_INSIGHTS_SRC` to point to the IIFE build, like following:

Reviewer: If they use an IIFE, can't they use `define`? I thought the problem was caused by the UMD boilerplate conflicting with Require.js.

Author: I want to avoid recommending it because when loading the lib via AMD, the library won't be able to capture interactions up until the point it's fully loaded.

Reviewer: Okay, so to clarify:

- Using the UMD build _without_ passing it to `define` crashes because of the anonymous `define` call in there.
- Using the UMD build by passing it to `define` crashes because there's no string ID.
- We want users to use a `<script>` tag and not rely at all on Require.js to load the library to make sure we start stacking events right away.

Is this accurate?

If yes, I would be straight about it in the explanation, right now we justify our solution (using an IIFE without `define`) with the wrong reason (using `define` would work and the initial error would no longer show up).

Author: All true except for this one: 

> Using the UMD build by passing it to define crashes because there's no string ID

Loading the UMD build of insights via `define` [works](https://codesandbox.io/s/busy-flower-fyuqt?file=/index.html:343-710), and recommanded by Require.js. But then you wouldn't be using the snippet.

> I would be straight about it in the explanation, right now we justify our solution (using an IIFE without define) with the wrong reason (using define would work and the initial error would no longer show up).

How about we add something like:
We recommand loading our library via our snippets, to garanty the capture interactions occuring before the libary is fully loaded. 
```suggestion
To work around this problem, avoid loading the library via a `define()` call. Instead, we recommand loading our library via the following snippets, to garanty the capture interactions occuring before the libary has been loaded. 
```

Reviewer: >Loading the UMD build of insights via define works, and recommended by Require.js. But then you wouldn't be using the snippet.

Okay, so it works but we recommend against it because it doesn't work well for us. So in this case, I'd just drop:

>To work around this problem, avoid loading the library via a `define()` call.

And instead go for:

```suggestion
To work around this problem and ensure you capture all interactions occurring before the library is done loading, change `ALGOLIA_INSIGHTS_SRC` to point to the IIFE build, and load it via a `<script>` tag.
```
